### PR TITLE
perf: use regexp in string escaping

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
Do you want some perf upgrades? I've got something for you.

### CODE REVIEW

**Comment by ivan-tymoshenko** [comment]
Снимок экрана 2022-10-15 в 15 22 11
Снимок экрана 2022-10-15 в 15 22 24

File: lib/serializer.js [hunk 1]
```diff
@@ -94,6 +97,10 @@ module.exports = class Serializer {
       str = str.toString()
     }
 
+    if (!STR_ESCAPE.test(str)) {
```
Comment (Uzlopak) [inline lib/serializer.js:100]: I was wondering, if following has a perf impact?
```suggestion
    if (STR_ESCAPE.test(str) === false) {
```

Could you please bench it?

File: lib/serializer.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
 'use strict'
 
+// eslint-disable-next-line
+const STR_ESCAPE = /[\u0000-\u001f\u0022\u005c\ud800-\udfff]|[\ud800-\udbff](?![\udc00-\udfff])|(?:[^\ud800-\udbff]|^)[\udc00-\udfff]/
```
Comment (Uzlopak) [inline lib/serializer.js:4]: Did you check with safe-regex2 package if it is a safe regex?

File: lib/serializer.js (hunk 1 above)
Comment (RafaelGSS) [reply lib/serializer.js:100]: It doesn't.

Comment (RafaelGSS) [inline lib/serializer.js:100]: Could you please include a comment above this line telling why it's needed?

File: lib/serializer.js (hunk 2 above)
Comment (ivan-tymoshenko) [reply lib/serializer.js:4]: It's safe.

File: lib/serializer.js (hunk 1 above)
Comment (Uzlopak) [reply lib/serializer.js:101]: Thx

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**